gateway/src/main/java/com/hurence/webapiservice/http/ingestion/ImportRequestParser.java

```diff
@@ -14,61 +16,49 @@
     private static final Logger LOGGER = LoggerFactory.getLogger(ImportRequestParser.class);
 
 
-    public String[] checkRequest (JsonArray metricsParam) {
-        final String[] error = new String[1];
-        try {
-            if (null == metricsParam) {
-                error[0] = "parameter array is NULL";
-                throw new IllegalArgumentException("parameter array is NULL");
-            }else if (metricsParam.isEmpty()) {
-                error[0] = "parameter array is empty";
-                throw new IllegalArgumentException("parameter array is empty");
+    public ResponseAndErrorHolder checkAndBuildValidHistorianImportRequest(JsonArray metricsParam) throws IllegalArgumentException {
+        // TODO should i keep this ?
+        /*if (null == metricsParam) {
+            throw new IllegalArgumentException("parameter array is NULL");
+        }else if (metricsParam.isEmpty()) {
+            throw new IllegalArgumentException("parameter array is empty");
+        }*/
```

Comment:
You may keep this, but modify message. "Empty request body" is better for example. Because "parameter array" does not mean much.

Reply:
done